core-api/src/main/java/com/optimizely/ab/odp/serializer/ODPJsonSerializerFactory.java

```diff
@@ -0,0 +1,49 @@
+/**
+ *    Copyright 2022, Optimizely Inc. and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.odp.serializer;
+
+import com.optimizely.ab.internal.JsonParserProvider;
+import com.optimizely.ab.odp.serializer.impl.GsonSerializer;
+import com.optimizely.ab.odp.serializer.impl.JacksonSerializer;
+import com.optimizely.ab.odp.serializer.impl.JsonSerializer;
+import com.optimizely.ab.odp.serializer.impl.JsonSimpleSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ODPJsonSerializerFactory {
+    private static final Logger logger = LoggerFactory.getLogger(ODPJsonSerializerFactory.class);
+
+    public static ODPJsonSerializer getSerializer() {
+        JsonParserProvider parserProvider = JsonParserProvider.getDefaultParser();
+        ODPJsonSerializer jsonSerializer = null;
+        switch (parserProvider) {
+            case GSON_CONFIG_PARSER:
+                jsonSerializer = new GsonSerializer();
+                break;
+            case JACKSON_CONFIG_PARSER:
```

Comment:
Its using the same JsonParserProvider to find out the default one. Its just exporting the implementation class separately. This is separate because it is a seralizer. I wanted to keep parser and serializers separate to avoid confusion. any more parsing or serialization done for ODP will go in to these same classes as added methods. Let me know what you think

Reply:
Separating parser and serializer looks good. I was talking about the future refactoring discussed already to reuse them in {datafile parser, event serializer, segments parser, and odp event serializer}.